src/seraphis_impl/enote_store.cpp

```diff
@@ -345,18 +408,39 @@ bool SpEnoteStore::try_import_legacy_key_image(const crypto::key_image &legacy_k
             "sp enote store (import legacy key image): intermediate and full legacy maps inconsistent (bug).");
 
         // c. set the full record
-        get_legacy_enote_record(record_ref->second.record,
-            legacy_key_image,
-            m_legacy_contextual_enote_records[legacy_enote_identifier].record);
-        events_inout.emplace_back(NewLegacyRecord{legacy_enote_identifier});
-
         // d. set the full record's contexts
-        update_contextual_enote_record_contexts_v1(
-                record_ref->second.origin_context,
-                spent_context,
-                m_legacy_contextual_enote_records[legacy_enote_identifier].origin_context,
-                m_legacy_contextual_enote_records[legacy_enote_identifier].spent_context
-            );
+        if (record_ref->second.is_type<LegacyContextualIntermediateEnoteRecordV1>())
+        {
+            m_legacy_contextual_enote_records[legacy_enote_identifier] = LegacyContextualEnoteRecordV1{};
```

Comment:
Before, when there was no `LegacyContextualEnoteRecordV1` in the map `m_legacy_contextual_enote_records` at the location `legacy_enote_identifier` we could initialize a new contextual record like this:
```
m_legacy_contextual_enote_records[legacy_enote_identifier];
```

Now `m_legacy_contextual_enote_records` stores the `LegacyContextualEnoteRecordVariant` and I think therefore we need to explicitly set it to the correct variant type like this:
```
m_legacy_contextual_enote_records[legacy_enote_identifier] = LegacyContextualEnoteRecordV1{};
```